src/pubsub/mod_pubsub_db_mnesia.erl

```diff
@@ -494,16 +524,33 @@ states_to_subscriptions([]) ->
 states_to_subscriptions([#pubsub_state{ subscriptions = [] } | RStates]) ->
     states_to_subscriptions(RStates);
 states_to_subscriptions([#pubsub_state{ stateid = {J, _}, subscriptions = Subs } | RStates]) ->
-    add_jid_to_subs(Subs, J, RStates).
+    add_jid_and_opts_to_subs(Subs, J, RStates).
 
--spec add_jid_to_subs(Subs :: [{mod_pubsub:subscription(), mod_pubsub:subId()}],
-                      LJID :: jid:ljid(),
-                      RStates :: [mod_pubsub:pubsubState()]) ->
-    [{jid:ljid(), mod_pubsub:subscription(), mod_pubsub:subId()}].
-add_jid_to_subs([], _J, RStates) ->
+-spec add_jid_and_opts_to_subs(Subs :: [{mod_pubsub:subscription(), mod_pubsub:subId()}],
+                               LJID :: jid:ljid(),
+                               RStates :: [mod_pubsub:pubsubState()]) ->
+    [{jid:ljid(), mod_pubsub:subscription(), mod_pubsub:subId(), mod_pubsub:subOptions()}].
+add_jid_and_opts_to_subs([], _J, RStates) ->
     states_to_subscriptions(RStates);
-add_jid_to_subs([{S, SubId} | RSubs], J, RStates) ->
-    [ {J, S, SubId} | add_jid_to_subs(RSubs, J, RStates) ].
+add_jid_and_opts_to_subs([{S, SubId} | RSubs], J, RStates) ->
+    Opts =
+    case mnesia:read({pubsub_subscription, SubId}) of
```

Comment:
This case and the one in line 549 look the same to me. Would it be worth to extract it to a function?